### PR TITLE
DList find by index

### PR DESCRIPTION
**Pull request by wallymathieu**

### CODE REVIEW

**Comment by wallymathieu** [comment]
It did not match the signature of find.

**Comment by wallymathieu** [comment]
By adding `findi`, `tryFindi` perhaps it makes sense to then also add `find` and `tryFind` ?

File: src/FSharpPlus/DList.fs [hunk 1]
```diff
@@ -76,6 +76,23 @@ type DList<'T>(length : int, data : DListData<'T> ) =
             | t::ts -> walk ts t xs
         walk [] l.dc state
 
+    static member findi (f : (int -> 'T -> bool)) (l:DList<'T>)  =
```
[the remark this answers is not in the document]
Comment (wallymathieu) [reply src/FSharpPlus/DList.fs]: yes, that could be a better choice

[the remark this answers is not in the document]
Comment (wallymathieu) [reply src/FSharpPlus/DList.fs]: I've marked it as private.

**Comment by gusty** [comment]
Yes, I was thinking the same. Also, they should be module functions ideally.
Marking as private is fine.

**Pull request merged**